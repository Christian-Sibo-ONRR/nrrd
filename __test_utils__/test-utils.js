import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@material-ui/core/styles'
import theme from '../src/js/mui/theme'
import { ApolloProvider } from '@apollo/react-hooks'

import { client } from '../src/apollo/client'
import {
  AppStatusProvider,
  DownloadProvider
} from '../src/stores'

import ErrorBoundary from '../src/components/ErrorBoundary'

const AllTheProviders = ({ children }) => (
  <ErrorBoundary>
    <ThemeProvider theme={theme}>
      <ApolloProvider client={client}>
        <AppStatusProvider>
          <DownloadProvider>
            {children}
          </DownloadProvider>
        </AppStatusProvider>
      </ApolloProvider>
    </ThemeProvider>
  </ErrorBoundary>
)
const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
