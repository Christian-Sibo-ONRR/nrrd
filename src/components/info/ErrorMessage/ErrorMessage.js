import React, { useContext } from 'react'

import { AppStatusContext } from '../../../stores/app-status-store'

import Modal from '@material-ui/core/Modal'
import Grid from '@material-ui/core/Grid'
import { makeStyles, useTheme } from '@material-ui/core/styles'

const useStyles = makeStyles(theme => ({
  backdrop: {
    zIndex: theme.zIndex.modal + 1,
    color: 'white',
  },
}))

const ErrorMessage = () => {
  const theme = useTheme()
  const classes = useStyles(theme)
  const { state } = useContext(AppStatusContext)

  const handleClose = () => {
    console.log('close')
  }
  return (
    <Modal open={state.ErrorMessage} onClose={handleClose} aria-labelledby="Error Message" aria-describedby="simple-modal-description">
      {state.ErrorMessage}
    </Modal>
  )
}

export default ErrorMessage
