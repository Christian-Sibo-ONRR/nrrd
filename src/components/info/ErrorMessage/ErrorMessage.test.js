/* eslint-disable no-undef */
import React from 'react'
import ErrorMessage from './ErrorMessage'

import '@testing-library/jest-dom/extend-expect'
import { render } from 'test-utils'

describe('Error Message:', () => {
  test('Found the text content', async () => {
    render(<ErrorMessage />)
    expect(true).toBeTruthy()
  })
})
