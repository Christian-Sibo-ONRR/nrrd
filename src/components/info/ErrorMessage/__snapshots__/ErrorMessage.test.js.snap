// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Message: Match snapshot 1`] = `null`;
